### DISCUSSION TITLE
WMF support on imagemagick 7.0.10

### DISCUSSION THREAD

**Discussion by PatelVishalJ** [ImageMagick/ImageMagick#4163] [Help]
I am installing imagemagick on apline, which picks up 7.0.10 version of imagemagick.

My primary use is to covert WMF to PNG. But `convert sample.wmf sample.png` gives error

```
convert: no decode delegate for this image format `WMF' @ error/constitute.c/ReadImage/572.
convert: no images defined `sample.png' @ error/convert.c/ConvertImageCommand/3322.
```
As per https://www.imagemagick.org/script/formats.php, I also installed libwmf, which does not resolve the issue.

`identify -list format | grep WMF` does not return any result.

Do I need additional packages installed?

**Comment by snibgo** [comment]
As you are using v7, I suggest you use "magick", not "convert".

> magick identify -list format | grep wmf does not return any result.

But the formats are listed in upper case like "WMF" not lower case like "wmf".

**Reply by PatelVishalJ** [reply]
@snibgo , thanks for the reply.
Either magick or convert gives me same error. 
`magick: no decode delegate for this image format 'WMF' @ error/constitute.c/ReadImage/572.`

"WMF" upper case `identify -list format | grep WMF` also not returning anything. I updated the question accordingly.

**Comment by snibgo** [comment]
Did you build IM yourself? If so, what does config.log say about WMF?

**Comment by fmw42** [comment]
You need to rebuild ImageMagick after installing libwmf or build libwmf into the RPM.

**Comment by PatelVishalJ** [comment]
@snibgo , @fmw42 ,
I did not build myself. I just did 
```
apk add libwmf
apk add imagemagick
```
is building by myself is the only option to make it work for wmf?

**Reply by fmw42** [reply]
Sorry, I do not use Linux, so cannot say what more you need.

**Comment by snibgo** [comment]
> is building by myself is the only option to make it work for wmf?

No. On Windows 8.1, I can read WMF fine, either with pre-built IM binaries or built myself (without needing libwmf).

I don't know what "apk add imagemagick" does. Does it download a pre-built binary, or build from source? Does it create log files?
